### ISSUE TITLE
Improvements on automatic wrapping

### ISSUE THREAD

**Issue by Kabouik** [mailpile/Mailpile#1862]
Automatic wrapping does not work on paragraphs starting with a "-", like lists, and the resulting email body looks weird because of paragraphs of different widths.

Additionally, I realized that wrapping is done every time the email is saved, _i.e._ if you are composing a message and leave it in your drafts for later editing. This is inconvenient because it means further edits on your draft will totally mess up the formatting of paragraphs. Here are pictures illustrating the issue:

*Composing a new mail (note paragraphs starting with "-"); no wrapping yet:*
![1](http://reho.st/https://0x0.st/0l7.png)

*Now wrapping has been done after leaving the "Compose" view, and then going back to the new draft (note the absence of wrapping for lists):*
![2](http://reho.st/https://0x0.st/0lh.png)

*Doing further edits to my email, the formatting starts to be messed up in the previously wrapped paragraphs, and makes editing more complicated (this email is short, but it may be very hard to read in longer drafts):*
![3](http://reho.st/https://0x0.st/0lF.png)

*Saving the draft, browsing to another Mailpile view, and then coming back to it will re-wrap the text as described above, but short lines in the previous screenshot may remain shorter than wrapping threshold (not sure on this one though, I have not properly tested it, and it is not obvious here, should try with more agressive editing before the save):*
![4](http://reho.st/https://0x0.st/0lC.png)

To avoid this issue, I believe wrapping should be done only when the content of the message is final and won't change anymore (when clicking on the `Send` button). Also, it would be nice to let the user specify the preferred length (in characters) for lines to apply automatic wrapping.

**TL;DR**
- Lists should be wrapped as well
- Wrapping should be done only when the content is not prone to further changes, _i.e._ when the email is actually sent and going in the outbox
- User preferences on wrapping could be convenient

**Comment by neurokrish** [comment]
Any update on this? I have the same issue as @Kabouik. Is there a temporary fix?

**Comment by JackDca** [comment]
Also relevant - #1638   #1560

On IRC  freenode.net - mailpile - ptotter referenced this which may be relevant.
https://useplaintext.email/#implementation-recommendations
